docs/topics/proc_creating-a-vertx-application.adoc

```diff
@@ -0,0 +1,204 @@
+
+* Write some simple application code that makes use of functionalities provided by {Vertx}.
+* Declare dependencies and configure the application build using a `pom.xml` file.
+* Start your application on localhost and verify that it works.
+
+
+.Prerequisites
+
+* Maven installed.
+
+.Procedure
+
+. Create the application directory and navigate to it.
++
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ mkdir myApp
+$ cd myApp
+----
+
+. Create a `pom.xml` file.
++
+[source,xml,options="nowrap",subs="attributes+"]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <name>My Application</name>
+  <description>Example application using RHOAR Vert.x</description>
+
+  <groupId>com.example</groupId>
+  <artifactId>my-app</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+  <packaging>jar</packaging>
+
+  <properties>
+    <vertx.version>3.4.2.redhat-009</vertx.version>
+    <vertx-maven-plugin.version>1.0.13</vertx-maven-plugin.version>
+    <vertx.verticle>io.openshift.booster.MyApplication</vertx.verticle>
+  </properties>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>io.vertx</groupId>
+        <artifactId>vertx-dependencies</artifactId>
+        <version>${vertx.version}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
+  <dependencies>
+    <dependency>
+      <groupId>io.vertx</groupId>
+      <artifactId>vertx-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.vertx</groupId>
+      <artifactId>vertx-web</artifactId>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>io.fabric8</groupId>
+        <artifactId>vertx-maven-plugin</artifactId>
+        <version>${vertx-maven-plugin.version}</version>
+        <executions>
+          <execution>
+            <id>vmp</id>
+            <goals>
+              <goal>initialize</goal>
+              <goal>package</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <redeploy>true</redeploy>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+
+  <profiles>
+    <profile>
+      <id>openshift</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>io.fabric8</groupId>
+            <artifactId>fabric8-maven-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>fmp</id>
+                <goals>
+                  <goal>resource</goal>
+                  <goal>build</goal>
+                </goals>
+              </execution>
+            </executions>
+            <configuration>
```

Comment:
Any reason for this configuration? Not that it's bad, but doesn't seem necessary.

Reply:
@Ladicek I don't think it's necessary for this particular example, either. I must have copy-pasted that from one of the POM files:) I'll double check if it's in any way necessary to have it in this example, and if not, I'll remove it.

Reply:
@Ladicek  After doing some light documentation-reading, I found that this particular configuration enables the [redeploy](https://vmp.fabric8.io/#how-does-the-redeploy-work) option for `mvn vertx:run` goal executed by the Vert.x Maven Plugin. The default value is `false`. IIUC, redeployment does not need to be enabled every time.

Reply:
Ah I had no idea FMP and the Vert.x Maven plugin interact like that, but it makes sense. (Somewhat :-) )